### PR TITLE
Added STS Auth support to Go Client

### PR DESCRIPTION
**Pull request by melanahammel**

### CODE REVIEW

**Comment by sdford** [comment]
Are you using `dep`? If so, you need a command in your Makefile that calls `dep ensure`.

Then, you call that Make command before your tests run.

You probably want to add a clean command too, that cleans up previous build directories.

File: README.md [hunk 1]
```diff
@@ -104,43 +102,33 @@ for _, v := range list {
 }
 ```
 
-For full information on every method, see the [Godoc]()
+For full information on every method, see the [Godoc]().
```
Comment (fieldju) [inline README.md]: Why is this link empty?

Comment (fieldju) [reply README.md]: Found the link, we should sanity check that there is automation that keeps these up to date.

Comment (melanahammel) [reply README.md]: GoDoc updates everyday and pulls the documentation from the comments for each function.

https://godoc.org/-/about

**Pull request merged**